### PR TITLE
Feature/loan reservation improvements

### PR DESCRIPTION
**Pull request by mian3010**

### CODE REVIEW

File: sites/all/modules/reol_reservation/reol_reservation.module [hunk 1]
```diff
@@ -211,17 +211,71 @@ function reol_reservation_get_reserve_button($entity) {
...
+  // Go through all messages, to remove old ones.
+  $all_messages = drupal_get_messages();
+  foreach ($all_messages as $level => $messages) {
+    foreach ($messages as $message) {
+      // If message is not the one set by this code, re-set it.
+      if (preg_match($d . $singular_string . $d, $message) == 0 && preg_match($d . $plural_string . $d, $message) == 0) {
+        drupal_set_message($message, $level);
+      }
+    }
+  }
```
Comment (xendk) [inline sites/all/modules/reol_reservation/reol_reservation.module]: What the hell is going on here?

Comment (mian3010) [inline sites/all/modules/reol_reservation/reol_reservation.module]: The problem is not trivial. When searching for something using the search bar, a 'modal' type popup appears that informs the user to wait. This triggers page_build, and sets the message.

When the page it actually shown page_build is triggered again, setting the message once more. The code above looks through the messages, to see if the message has been set before. If it has, it removes it, as we are about to set it again.

Comment (mian3010) [inline sites/all/modules/reol_reservation/reol_reservation.module]: I have added this as a comment before doing the magic.

Comment (xendk) [inline sites/all/modules/reol_reservation/reol_reservation.module]: Doesn't setting $repeat = FALSE fix the problem?

https://api.drupal.org/api/drupal/includes%21bootstrap.inc/function/drupal_set_message/7

Comment (mian3010) [inline sites/all/modules/reol_reservation/reol_reservation.module]: Only in the case that the number of reservations stay the same at each page load between displaying messages. But that would be the case for 99% of the time. I will change it.

**Pull request merged**